src/Files.App/Views/MainPage.xaml.cs

```diff
@@ -188,7 +188,8 @@ public void MultitaskingControl_CurrentInstanceChanged(object? sender, CurrentIn
 			UpdateStatusBarProperties();
 			UpdateNavToolbarProperties();
 			LoadPaneChanged();
-			NavigationHelpers.UpdateInstancePropertiesAsync(navArgs);
+			UpdateLayoutIfRequired(e.CurrentInstance);
+			await NavigationHelpers.UpdateInstancePropertiesAsync(navArgs);
```

Comment:
@hishitetsu does this need to be awaited?

Reply:
It is not necessary, but it is easier to debug. But if you are going to await, this should be done at the end of the method to prevent subsequent processing delays.